grails-app/jobs/criaapp/TaskDueDateJob.groovy

```diff
@@ -0,0 +1,29 @@
+package criaapp
+
+import com.asinha.domain.Payment
+import com.asinha.domain.PaymentService
+import com.asinha.enums.PaymentStatus
+import com.asinha.utils.CustomDateUtils
+
+import grails.gorm.transactions.Transactional
+
+@Transactional
+class TaskDueDateJob {
+  
+    static triggers = {
+        cron name: 'dueDate', cronExpression: "0 0/1 * 1/1 * ? *"
+    }
+  
+    static concurrent = false
+
+    def paymentService
+
+    def execute(){
+        Date yesterday = CustomDateUtils.getYesterday()
+        List<Payment> paymentList = paymentService.listPaymentByStatusAndDate(PaymentStatus.PENDING, yesterday)
```

Comment:
Cria um método no paymentService que faça essa parte de de buscar a paymentList e fazer o for para atualizar o status. Torna a responsabilidade do teu job de apenas chamar um método la do service, sem trazer regras para ele